src/caja-window-slot.c

```diff
@@ -384,34 +384,43 @@ caja_window_slot_update_icon (CajaWindowSlot *slot)
     icon_name = NULL;
     if (info)
     {
+        GtkWindow *gtk_window;
+        GdkPixbuf *pixbuf;
+
+        gtk_window = GTK_WINDOW (window);
+
+        pixbuf = gtk_window_get_icon (gtk_window);
+        if (pixbuf)
+            gtk_window_set_icon (gtk_window, NULL);
```

Comment:
little question, we need `g_object_unref (pixbuf);` inside this condition?

Reply:
I added a comment in the code. You don't need to call g_object_unref, calling gtk_window_set_icon or gtk_window_set_icon_list frees the icon/list when the new value is NULL.